scintillometry/utils/phase_utils.py

```diff
@@ -0,0 +1,142 @@
+                                  bipm_version=bipm_version,
+                                  include_gps=include_gps,
+                                  planets=planets, tdb_method=tdb_method,
+                                  **kwargs)
+        self.obs = obs
+        self.frequency = frequency
+
+    def __call__(self, t):
+        """Compute the apparent phase at one or more time stemp
+
+           Parameter
+           ---------
+           t : `astropy.time.Time` instance
+               The input time stamps.
+
+           Return
+           ------
+           The apparent pulse phase at time t. The phases are return as
+           'integer phase, fractional phase'
+        """
+        toas = self._make_toas(t)
+        ph = self.model.phase(toas)
+        return (u.Quantity(ph.int, u.cycle, copy=False),
+                u.Quantity(ph.frac, u.cycle, copy=False))
+
+    def _make_toas(self, t):
+        toa_list = make_toa_list(t, self.obs, self.frequency)
+        toas = self.toa_maker.make_toas(toa_list=toa_list)
+        return toas
+
+    def apparent_spin_freq(self, t):
+        """Compute the apparent spin frequency at one or more time stamps.
+
+            Parameter
+            ---------
+            t : `astropy.time.Time` instance
+                The input time stamps.
+
+            Return
+            ------
+            The apparent spin frequency at time t.
+        """
+        toas = self._make_toas(t)
+        return self.model.d_phase_d_toa(toas)
+
+
+class PolycoPhase(object):
+    """A utility class for a Tempo style polyco phase calculation.
+
+       Parameter
+       ---------
+       polyco_file : str
+           Tempo style polyco file.
+    """
+    def __init__(self, polyco_file):
+        self.polyco = Polyco(polyco_file)
+
+    def __call__(self, t):
+        """Compute the apparent phase at one or more time stemp
+
+           Parameter
+           ---------
+           t : `astropy.time.Time` instance
+               The input time stamps.
+
+           Return
+           ------
+           The apparent pulse phase at time t. The phases are return as
+           'integer phase, fractional phase'.
+        """
+        t = t._apply(np.atleast_1d)
+        ph = self.polyco(t)
+        with u.set_enabled_equivalencies([(u.cycle, None)]):
+            ph_int, ph_frac = divmod(ph, 1)
+            return (u.Quantity(ph_int, u.cycle, copy=False),
+                    u.Quantity(ph_frac, u.cycle, copy=False))
+
+    def apparent_spin_freq(self, t):
+        """Compute the apparent spin frequency at one or more time stamps.
+
+            Parameter
+            ---------
+            t : `astropy.time.Time` instance
+                The input time stamps.
+
+            Return
+            ------
+            The apparent spin frequency at time t.
+        """
+        t = t._apply(np.atleast_1d)
+        apprnt_f0 = self.polyco(t, deriv=1)
```

Comment:
Still dislike this half-abbreviation. Call it `f` or just `return self.polyco(...).to(...)`